example/lib/main.dart

```diff
@@ -183,85 +187,20 @@ class _MyAppState extends State<MyApp> {
-    Stream<ably.ConnectionStateChange> stream = await realtime.connection.on();
-    StreamSubscription<ably.ConnectionStateChange> omegaSubscription;
-    omegaSubscription = stream.listen((ably.ConnectionStateChange stateChange) async {
-      print('DISPOSABLE LISTENER ω :: Change event arrived!: ${stateChange.event}');
-      if (stateChange.event == ably.ConnectionEvent.connected) {
-        await omegaSubscription.cancel();
-      }
-    });
-
-    //RETAINING LISTENER - β
-    StreamSubscription<ably.ConnectionStateChange> betaSubscription = realtime.connection.on().listen((ably.ConnectionStateChange stateChange) async {
-      print('RETAINING LISTENER β :: Change event arrived!: ${stateChange.event}');
-      // NESTED LISTENER - ξ
-      // will be registered only when connected event is received by β listener
-      StreamSubscription<ably.ConnectionStateChange> etaSubscription = realtime.connection.on().listen((
-        ably.ConnectionStateChange stateChange) async {
-        // k ξ listeners will be registered
-        // and each listener will be called `n-k` times respectively
-        // if listener β is called `n` times
-        print('NESTED LISTENER ξ: ${stateChange.event}');
-      });
-      connectionStateChangeSubscriptions.add(etaSubscription);
-    });
-
-    StreamSubscription<ably.ConnectionStateChange> preZetaSubscription;
-    StreamSubscription<ably.ConnectionStateChange> postZetaSubscription;
-    preZetaSubscription = realtime.connection.on().listen((ably.ConnectionStateChange stateChange) async {
-      //This listener "pre ζ" will be cancelled from γ
-      print('NESTED LISTENER "pre ζ": ${stateChange.event}');
-    });
-
-
-    //RETAINING LISTENER - γ
-    StreamSubscription<ably.ConnectionStateChange> gammaSubscription = realtime.connection.on().listen((ably.ConnectionStateChange stateChange) async {
-      print('RETAINING LISTENER γ :: Change event arrived!: ${stateChange.event}');
-      if (stateChange.event == ably.ConnectionEvent.connected) {
-        await preZetaSubscription.cancel();  //by the time this cancel is triggered, preZeta will already have received current event.
-        await postZetaSubscription.cancel(); //by the time this cancel is triggered, postZeta hasn't received the event yet. And will never receive as it is cancelled.
-      }
-    });
-
-    postZetaSubscription = realtime.connection.on().listen((ably.ConnectionStateChange stateChange) async {
-      //This listener "post ζ" will be cancelled from γ
-      print('NESTED LISTENER "post ζ": ${stateChange.event}');
-    });
-
-    connectionStateChangeSubscriptions = [
-      alphaSubscription,
-      betaSubscription,
-      gammaSubscription,
-      omegaSubscription,
-      preZetaSubscription,
-      postZetaSubscription
-    ];
-
-    setState(() {
-      _realtime = realtime;
-      _realtimeCreationState = OpState.Succeeded;
-    });
+    connectionStateChangeSubscriptions = [ alphaSubscription ];
```

Comment:
This looks brittle to me. 

I'd instead change the field
```dart
List<StreamSubscription<ably.ConnectionStateChange>> connectionStateChangeSubscriptions;
```
to 
```
var _connectionStateChangeSubscriptions = <StreamSubscription<ably.ConnectionStateChange>>[];
```
and this code to 
```dart
_connectionStateChangeSubscriptions.add(alphaSubscription);
```

Reply:
addressed 86c4e6b0175a59c09dfd368dcc6130365c9f342d